include/TS3.h

```diff
@@ -3,27 +3,36 @@
 
 #include <iostream>
 
-#include "TDetector.h"
+#include "TGRSIDetector.h"
 #include "TS3Hit.h"
 
-class TS3 : public TDetector {
+class TS3 : public TGRSIDetector {
 	public:
 		TS3();
-		~TS3();
+		TS3(const TS3&);
+		virtual  ~TS3();
 
-		void AddFragment(TFragment*, MNEMONIC*);
-		void BuildHits() {} //no need to build any hits, everything already done in AddFragment
+		virtual void AddFragment(TFragment*, MNEMONIC*);
+		virtual void BuildHits();
 
+		TGRSIDetectorHit* GetHit(const int& idx =0);
 		TS3Hit* GetS3Hit(const int& i);  
-		Short_t GetMultiplicity() { return fS3Hits.size(); }
+		Short_t GetMultiplicity() const { return fS3Hits.size(); }
+		void PushBackHit(TGRSIDetectorHit* deshit);
 
-		TVector3 GetPosition(int front, int back);
+		static TVector3 GetPosition(int ring, int sector);
 
-      virtual void Clear(Option_t *opt = "all");		     //!
-      virtual void Print(Option_t *opt = "") const;		  //!
+		void Copy(TObject&) const;
+		TS3& operator=(const TS3&);  // 
+		virtual void Clear(Option_t *opt = "all");		     //!
+		virtual void Print(Option_t *opt = "") const;		  //!
+		
+		
 
 	private:
 		std::vector<TS3Hit> fS3Hits;
+		std::vector<TFragment*> fS3_RingFragment; //! 
+		std::vector<TFragment*> fS3_SectorFragment; //! 
```

Comment:
As far as build hits go, I think you should like at the time difference
between combinations.  In scenarios like the csm experiments when we are
detecting to simultaneous alphas at the same time to get the physics, this
has done the trick.  As the spice detector system this should be more than
sufficient to do the trick.  One should be able to set such timing gates
from info stored in the fragment tree.  Just for my own curiosity, I assume
you guys are no longer doing colex experiments with spice since you are
worried about multiplicity, so that means fusion evaporation? Are there
particular channels you are looking for?




On Wed, Nov 4, 2015 at 8:48 PM, jsmallcombe <notifications@github.com>
wrote:

> In include/TS3.h
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584#discussion_r43967838>
> :
>
> >
> >  	private:
> >  		std::vector<TS3Hit> fS3Hits;
> > +		std::vector<TFragment*> fS3_RingFragment; //!
> > +		std::vector<TFragment*> fS3_SectorFragment; //!
>
> More than 2 hits (4 combinations) should be rare.
>
> For now I would leave it as is (with //!) this is the same as how it was
> before the removal of the data class, but faster for the build as it doesnt
> involve the double copy of the data class.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GRIFFINCollaboration/GRSISort/pull/584/files#r43967838>
> .
>